modules/JournalControl/src/journalmodelcollection.cpp

```diff
@@ -226,7 +226,7 @@ int JournalModelCollection::dumpToFile(std::string fileName) {
 
 				if (section.end - section.beg < 0) {
 					RCLCPP_ERROR(get_logger(), "End precedes beginning in file %s: beg @%ld, end @%ld",
-							   file.c_str(), section.beg, section.end);
+							   file.c_str(), (long int)section.beg, (long int)section.end);
```

Comment:
Maybe better to change the type in logging output from %ld to whatever fits section.beg?
Instead of casting the values to fit %ld?

Ignore this comment if irrelevant

Reply:
Yes I would also like to do that, but in my understanding what type `section.beg` and `section.end` are depends on the system, so we can't know if it should be `%ld` or any other value. So casting this to a long int should be safe I hope.